### PR TITLE
Update release guide for maintainers

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Documentation for https://github.com/iterative/cml/pull/997

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
Note that the whole release process can be initiated from the web user interface, by visiting the [workflow page](https://github.com/iterative/cml/actions/workflows/release.yml) and [running it](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow).

File: content/docs/contributing/core.md [hunk 1]
```diff
@@ -71,10 +71,9 @@ you experience any problems, please don't hesitate to ping us in our
...
 - **New releases**
-  - `git checkout master && git pull && git checkout -b M.m.p && npm version M.m.p && gh pr create`
-  - Merge the resulting PR
-  - `git checkout master && git pull && git tag -f vM.m.p && git push --tags` or
-    comment `/tag vM.m.p SHA` in the PR
-  - Wait for a draft to appear under
-    [releases](https://github.com/iterative/cml/releases)
-  - Check & publish the draft
+  - Run `gh workflow run release.yml --field bump=<increment>` where
+    `<increment>` is either `patch`, `minor` or `major`
```
Comment (casperdcl) [inline content/docs/contributing/core.md:75]:
```suggestion
  - `gh workflow run release.yml --field bump={patch,minor,major}`
```

File: content/docs/contributing/core.md [hunk 2]
```diff
@@ -71,10 +71,9 @@ you experience any problems, please don't hesitate to ping us in our
...
+  - Wait for an automatically created pull request to appear and approve it; it
+    will be merged as soon as checks succeed
```
Comment (casperdcl) [inline content/docs/contributing/core.md:77]:
```suggestion
  - Wait for a PR to appear
  - Check & approve the PR
```

File: content/docs/contributing/core.md [hunk 3]
```diff
@@ -71,10 +71,9 @@ you experience any problems, please don't hesitate to ping us in our
...
+  - Wait for the new release to appear, and customize the release notes if
+    required
```
Comment (casperdcl) [inline content/docs/contributing/core.md:79]: needs to be `--draft`

```suggestion
  - Wait for a draft to appear under
    [releases](https://github.com/iterative/cml/releases)
  - Check & publish the draft
```

Comment (0x2b3bfa0) [reply content/docs/contributing/core.md:78]: That requires more human interaction 🤔

Comment (casperdcl) [reply content/docs/contributing/core.md:78]: `--draft`: https://github.com/iterative/cml/pull/1165 :)

**Pull request merged**